homeassistant/components/sia/sia_entity_base.py

```diff
@@ -2,51 +2,81 @@
 from __future__ import annotations
 
 from abc import abstractmethod
+from dataclasses import dataclass
 import logging
-from typing import Any
 
 from pysiaalarm import SIAEvent
 
-from homeassistant.config_entries import ConfigEntry
-from homeassistant.const import CONF_PORT
 from homeassistant.core import CALLBACK_TYPE, State, callback
 from homeassistant.helpers.dispatcher import async_dispatcher_connect
-from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity import DeviceInfo, EntityDescription
 from homeassistant.helpers.event import async_call_later
 from homeassistant.helpers.restore_state import RestoreEntity
+from homeassistant.helpers.typing import StateType
 
-from .const import CONF_ACCOUNT, CONF_PING_INTERVAL, DOMAIN, SIA_EVENT, SIA_NAME_FORMAT
+from .const import DOMAIN, SIA_EVENT, SIA_HUB_ZONE, SIA_NAME_FORMAT, SIA_NAME_FORMAT_HUB
 from .utils import get_attr_from_sia_event, get_unavailability_interval
 
 _LOGGER = logging.getLogger(__name__)
 
 
+def get_name(
+    port: int, account: str, zone: int | None, device_class: str | None
+) -> str:
+    """Return the name of the zone."""
+    if not device_class:
+        device_class = "none"
+    if zone is None or zone == 0:
+        return SIA_NAME_FORMAT_HUB.format(port, account, device_class)
+    return SIA_NAME_FORMAT.format(port, account, zone, device_class)
+
+
+@dataclass
+class SIAEntityDescription(EntityDescription):
+    """Required keys for SIA entities."""
+
+    code_consequences: dict[str, StateType | bool] = {}
```

Comment:
The default value for `code_consequences` seems to be compulsory (all the descriptions provide one).
It should be moved to a `RequiredKeysMixIn`:

```python
@dataclass
class SIARequiredKeysMixIn():
    """Required keys for SIA entities."""

    code_consequences: dict[str, StateType | bool]


@dataclass
class SIAEntityDescription(SIARequiredKeysMixIn, EntityDescription):
    """Base entity description for SIA entities."""

```